system-addon/content-src/activity-stream-prerender.jsx

```diff
@@ -0,0 +1,63 @@
+const React = require("react");
+const ReactDOM = require("react-dom/server");
+const Base = require("content-src/components/Base/Base");
+const {Provider} = require("react-redux");
+const initStore = require("content-src/lib/init-store");
+const {reducers, INITIAL_STATE} = require("common/Reducers.jsm");
+const {actionTypes: at} = require("common/Actions.jsm");
+const allStrings = require("../data/locales.json");
+const EMPTY_LOCALE = "en-PRERENDER";
+
+function prerender(locale) {
+  const store = initStore(reducers, INITIAL_STATE);
+  let strings = {};
+
+  if (locale) {
+    if (!allStrings[locale]) {
+      throw new Error(`Tried to get strings for ${locale} but none were found.`);
+    }
+    strings = allStrings[locale];
+  } else {
+    Object.keys(allStrings["en-US"]).forEach(key => {
+      strings[key] = " ";
+    });
+  }
```

Comment:
@Mardak, does this need a spin-off bug to properly do individual string fallback correctly?

Reply:
I think string fallbacks work decently well at the moment (they fall back to English for any missing strings), although there are some cases where dynamic string ids get injected (e.g. TopStories) that could be handled better on a case-by-case basis